enterprise_access/apps/subsidy_access_policy/models.py

```diff
@@ -350,7 +358,7 @@ def clean(self):
         """
         Used to help validate field values before saving this model instance.
         """
-        if self.is_spend_limit_updated:
+        if self.active and (self.is_active_updated or self.is_spend_limit_updated):
```

Comment:
Modified the conditional to avoid the path of a user disabling the policy. Setting an out of bounds spend_limit, then re-enabling the policy.
Before conditional added:
https://github.com/openedx/enterprise-access/assets/82611798/16a80f8d-de14-43ac-8ba3-efd71f9e5751


https://github.com/openedx/enterprise-access/assets/82611798/16a80f8d-de14-43ac-8ba3-efd71f9e5751